### ISSUE TITLE
Auto-create necessary directories during writes when using ASSET_STORAGE_TYPE=filesystem

### ISSUE THREAD

**Issue by Southclaws** [Southclaws/storyden#332]
for #321 when `ASSET_STORAGE_TYPE` is set to empty (local filesystem mode) this code doesn't appear to work:

```go

func NewLocalStorer(cfg config.Config) Storer {
	var path string
	if cfg.AssetStorageLocalPath != "" {
		path = cfg.AssetStorageLocalPath
	} else {
		path = "./data"
	}

	if err := os.MkdirAll(path, 0o755); err != nil {
		panic(err)
	}
	if err := os.MkdirAll(filepath.Join(path, "assets"), 0o755); err != nil {
		panic(err)
	}

	s := os.DirFS(path)

	return &localStorer{s, path}
}

```

**Comment by Southclaws** [comment]
it might be better to check for this folder during writes and create it then, cfg.AssetStorageLocalPath might be interpreted wrongly or `./data` may be not where we think it is (maybe not root `/` when running in the container)

needs tests

**Comment by Southclaws** [comment]
also no idea why i wrote two mkdirall calls here... probably needs a rewrite, note to self...

seems this PR introduced the code: https://github.com/Southclaws/storyden/pull/49

**Comment by Southclaws** [comment]
fixed in b0299610bbecf2e9bc4096484ef6fa9038ebe65d
